packages/workspace/src/utilities/typescript/compilation.ts

```diff
@@ -13,9 +14,16 @@ export interface TypeScriptCompilationOptions {
   watch?: boolean;
 }
 
+export interface TypescriptWatchChangeEvent {
+  diagnostic: ts.Diagnostic;
+  newLine: string;
+  options: ts.CompilerOptions;
+  errorCount: number;
+}
+
 export function compileTypeScript(
   options: TypeScriptCompilationOptions
-): { success: boolean } | Promise<any> {
+): { success: boolean } | Observable<TypescriptWatchChangeEvent> {
```

Comment:
I think it would be better to split this into 2 different methods.

1. For watch mode that always returns an observable (or an async generator if possible)
2. For not watch mode that returns synchronously. (Assuming there's no async version)